hatch_pip_compile/plugin.py

```diff
@@ -166,32 +180,42 @@ def _pip_sync_cli(self) -> None:
             "--verbose" if self.config.get("pip-compile-verbose", None) is True else "--quiet",
             "--python-executable",
             str(self.virtual_env.python_info.executable),
-            str(self._piptools_lock_file),
         ]
         if not self.dependencies:
             self._piptools_lock_file.write_text("")
+        extra_args = self.config.get("pip-compile-install-args", [])
+        cmd.extend(extra_args)
+        cmd.append(str(self._piptools_lock_file))
         self.virtual_env.platform.check_command(cmd)
         if not self.dependencies:
             self._piptools_lock_file.unlink()
+        if not self.skip_install:
+            if self.dev_mode:
+                self.install_project_no_deps_dev()
+            else:
+                self.install_project_no_deps()
```

Comment:
It seems like for the case of pip-sync, `install_project*` is executed twice in our own code which we have full control over.

* `_hatch_pip_compile_install`:
   * calls `_piptools_install_dependencies`:
       * calls `_pip_sync_cli`:
          * calls `install_project*`
   * calls `install_project*`


```suggestion
```

Reply:
When `pip-sync` is run on a lockfile it uninstalls the project from the environment, since the lockfile only has third party dependencies - so the project needs to be re reinstalled each time.

Reply:
OK but if you look at the list above *flattened*, this is the sequence of actions at the moment:

1. `_pip_sync_cli`
2. `install_project_no_deps*`
3. `install_project_no_deps*`

I'm saying that surely one of the two could be dropped

Reply:
Ah, you're right. Thanks for breaking that down.